### ISSUE TITLE
Double underscore instead of dots

### ISSUE THREAD

**Issue by geirivarjerstad** [appurist/vscode-json-flattener#3]
Hi, I would like to request using the option to use double underscore "__" instead of dots "."

The need comes from flattening dotnet appsettings.json files where "__" is used instead of dots.

Example:

```
  "Logging__LogLevel__Microsoft": "Debug",
  "Logging__LogLevel__Default": "Debug",
```

Kind regards, Geir

**Comment by appurist** [comment]
Sorry, just saw this catching up on emails. That sounds very reasonable and I'll take a look at that as soon as I can.

Do you have any suggestion for how you'd like to see this triggered, as an option? I suppose I could just provide another pair of handers that use a different separator, that you could use instead.

The only other caveat is that I do have a plan for _unflatten_ eventually too, and it may not be possible to reliably unflatten identifiers using `__` (although it might only be a problem in practice where developers have used leading underscores, such as `__FILE__` or `__DATE__,` and those could be avoided).

**Comment by appurist** [comment]
@geirivarjerstad I haven't done any dotnet work in years; is the file in question always `appsettings.json`? Or would there be other cases for assuming `__` rather than `.`?  I'm trying to determine how to know _**when**_ to assume `__` rather than `.` for the separator, without creating separate commands.

**Comment by appurist** [comment]
Sorry @geirivarjerstad  I misunderstood; I'm still thinking about _unflatten_.  I believe your request is actually to change the _output_ to use `__` as the generated separator, not to recognize it on input (unflatten). 

I think we'll need a separate command for that although it _could_ auto-recognize `appsettings.json` specifically and behave differently for that one filename. But it seems inappropriate though to reserve a specific filename, so a separate VS Code command is probably best.

**Comment by appurist** [comment]
I have released a 1.0.6 version (available now at https://marketplace.visualstudio.com/items?itemName=Appurist.json-flattener)  that includes a 5th command, a "`Flatten (Selection - dotnet__separator)`" with an id of `appurist.json-flattener.selection-dotnet` which you can find with Ctrl-Shift-P as usual. Update and this command should allow you to select json text in an editor and convert it to the flattened format with dotnet `__` separators.

Feel free to comment or reopen if this isn't quite what this Issue was looking for.
